ethier/sign.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/divergencetech/ethier/eth"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/spf13/cobra"
+)
+
+var signer *eth.Signer
+
+func init() {
+	var signCmd = &cobra.Command{
+		Use:   "sign",
+		Short: "Signs messages from stdin using an ECDSA signer.",
+	}
+
+	rootCmd.AddCommand(signCmd)
+	signCmd.PersistentFlags().String("mnemonic", "", "Path to file containing the mnemonic, password and account number (newline separated) for key derivation")
+
+	var signAddrCmd = &cobra.Command{
+		Use:     "addresses",
+		Short:   "Signs addresses from stdin using an ECDSA signer.",
+		PreRunE: getSigner,
+		RunE:    signAddresses,
+	}
+
+	signCmd.AddCommand(signAddrCmd)
+}
+
+type SignedAddress struct {
+	Address   string `json:"address"`
+	Signature string `json:"signature"`
+}
+
+func getSignerFromMnemonic(filepath string) (*eth.Signer, error) {
+	dat, err := os.ReadFile(filepath)
+	if err != nil {
+		return nil, fmt.Errorf("Read mnemonic: %v", err)
+	}
+
+	lines := strings.Split(string(dat), "\n")
+
+	mnemonic := lines[0]
+	var password string = ""
+	var account uint = 0
+
+	if len(lines) > 1 {
+		password = lines[1]
+	}
+
+	if len(lines) > 2 {
+		if lines[2] != "" {
+			acc, err := strconv.Atoi(lines[2])
+			if err != nil {
+				return nil, fmt.Errorf("Converting derivation account: %v", err)
+			}
+			account = uint(acc)
+		}
+	}
+
+	signer, err := eth.DefaultHDPathPrefix.SignerFromSeedPhrase(
+		mnemonic, password, account,
+	)
+
+	if err != nil {
+		return nil, fmt.Errorf("Generate signer: %v", err)
+	}
+	return signer, nil
+}
+
+func getSigner(cmd *cobra.Command, args []string) (retErr error) {
+	mnemonicPath, err := cmd.Flags().GetString("mnemonic")
+	if err != nil {
+		log.Fatalf("Get mnemonics flag: %v", err)
+	}
+
+	if mnemonicPath != "" {
+		signer, err = getSignerFromMnemonic(mnemonicPath)
+		if err != nil {
+			log.Fatalf("Generate signer from mnemonic: %v", err)
+		}
+	} else {
+		signer, err = eth.NewSigner(256)
+		if err != nil {
+			log.Fatalf("Generate new signer: %v", err)
+		}
+	}
+
+	return nil
+}
+
+// sign addresses generates a new signer or derives it from a given mnemonic
+// to sign a list of addresses
+func signAddresses(_ *cobra.Command, args []string) (retErr error) {
+	defer func() {
+		if retErr != nil {
+			retErr = fmt.Errorf("signing: %w", retErr)
+		}
+	}()
+
+	buf, err := io.ReadAll(os.Stdin)
+	if err != nil {
+		log.Fatalf("Read stdin: %v", err)
```

Comment:
`return fmt.Errorf("read addresses on stdin: %v", err)`

Note lower-case message: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Same applies everywhere that you've used `log.Fatalf()`, which should really only be used in `init()` functions and when inside a function that is forced to have a signature that doesn't return an error (e.g. for integration with someone else's API). There's no need to stop an entire binary just because of a single error even if you know that it will inevitably be the case at this point in time, plus `Fatalf` prints an entire stack trace, which is horrible for end users.